openedge-plugin/src/main/java/org/sonar/plugins/openedge/sensor/OpenEdgeXREFSensor.java

```diff
@@ -92,15 +95,18 @@ public void execute(SensorContext context) {
       ruleTime.put(entry.getKey().ruleKey().toString(), 0L);
     }
 
+    if (settings.useXrefFilter()) {
+      LOG.info("XML XREF filter activated");
+    }
     for (InputFile file : fileSystem.inputFiles(fileSystem.predicates().hasLanguage(Constants.LANGUAGE_KEY))) {
       LOG.debug("Looking for XREF of {}", file.relativePath());
 
       File xrefFile = getXrefFile(file.file());
       if ((xrefFile != null) && xrefFile.exists()) {
         LOG.debug("Parsing XML XREF file {}", xrefFile.getAbsolutePath());
-        try {
+        try (InputStream inpStream = new FileInputStream(xrefFile)) {
           long startTime = System.currentTimeMillis();
-          Document doc = dBuilder.parse(xrefFile);
+          Document doc = dBuilder.parse(settings.useXrefFilter() ? new InvalidXMLFilterStream(inpStream) : inpStream);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Convert this usage of the ternary operator to an "if"/"else" structure. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1774)